edx_proctoring/static/proctoring/js/views/proctored_exam_onboarding_view.js

```diff
@@ -0,0 +1,247 @@
+        proctoring_started: gettext('Proctoring Started'),
+        started: gettext('Started'),
+        submitted: gettext('Submitted'),
+        verified: gettext('Verified'),
+        rejected: gettext('Rejected'),
+        error: gettext('Error')
+    };
+    viewHelper = {
+        getDateFormat: function(date) {
+            if (date) {
+                return new Date(date).toString('MMM dd, yyyy h:mmtt');
+            } else {
+                return '---';
+            }
+        },
+        getOnboardingStatus: function(status) {
+            if (onboardingStatuses.includes(status)) {
+                return onboardingStatusReadableFormat[status];
+            } else {
+                return status;
+            }
+        }
+    };
+    edx.instructor_dashboard.proctoring.ProctoredExamOnboardingView = Backbone.View.extend({
+        initialize: function() {
+            this.setElement($('.student-onboarding-status-container'));
+            this.collection = new edx.instructor_dashboard.proctoring.ProctoredExamOnboardingCollection();
+            this.templateUrl = '/static/proctoring/templates/student-onboarding-status.underscore';
+            this.courseId = this.$el.data('course-id');
+            this.template = null;
+
+            this.initialUrl = this.collection.url;
+            this.collection.url = this.initialUrl + this.courseId;
+            this.inSearchMode = false;
+            this.searchText = '';
+            this.filters = [];
+            this.currentPage = 1;
+            this.data = {};
+
+            /* re-render if the model changes */
+            this.listenTo(this.collection, 'change', this.collectionChanged);
+
+            /* Load the static template for rendering. */
+            this.loadTemplateData();
+        },
+        events: {
+            'click .search-onboarding > span.search': 'searchItems',
+            'click .search-onboarding > span.clear-search': 'clearSearch',
+            'click .status-checkboxes > li > input': 'filter',
+            'click li > a.target-link': 'getPaginatedAttempts'
+        },
+        searchItems: function(event) {
+            var searchText = $('#search_onboarding_id').val();
+            if (searchText !== '') {
+                this.inSearchMode = true;
+                this.searchText = searchText;
+                this.currentPage = 1;
+                this.collection.url = this.constructUrl();
+                this.hydrate();
+                event.stopPropagation();
+                event.preventDefault();
+            }
+        },
+        clearSearch: function(event) {
+            this.inSearchMode = false;
+            this.searchText = '';
+            this.currentPage = 1;
+            this.collection.url = this.constructUrl();
+            this.hydrate();
+            event.stopPropagation();
+            event.preventDefault();
+        },
+        filter: function(event) {
+            var $checkbox = $(event.currentTarget);
```

Comment:
Can you add some comments to this function on what it's general structure and purpose is? For exam, this function is invoked every time user clicks on the check box to filter based on onboarding status. And the function would refresh the view based on the records returned from backend API.

Reply:
I changed this to include submit/clear buttons, rather than filtering every time a checkbox is clicked.